### ISSUE TITLE
Plugin attempts to retrieve MFA with every run

### ISSUE THREAD

**Issue by earchibald-lv** [xeger/awsume-1password-plugin#1]
**Describe the bug**
Plugin attempts to retrieve the MFA with each run. The MFA is still cached by awsume, and if I cancel the plugin auth awsume still functions properly,

**Expected behavior**
The plugin should not request 1Password auth if the MFA is cached by awsume

**Output**
Invoke `awsume` with the `--debug` flag and paste the resulting output below. Make sure to include the command you ran, with all of the flags. If the program crashed then make sure to include the stack trace and all relevant error messages.

```
❯ awsume sre-team@lv-tst2
1Password: error initializing client: authorization prompt dismissed, please try again

Session token will expire at 2023-08-29 20:58:35
[sre-team@lv-tst2] Role credentials will expire 2023-08-29 16:03:39
```

**Configuration**
If relevant, please share your AWSume configuration file below. This file should be free of secrets, but please verify before submitting your report.

```yaml
1password: aws_lv-techops_earchibald
colors: true
console:
  browser:
    command: ''
  browser_command: ''
fuzzy-match: true
role-duration: 0
role-session-name: earchibald
```

**Additional context**

**Comment by xeger** [comment]
I see the problem, @earchibald-lv; I'll look into it. Thanks for the bug report.

**Comment by xeger** [comment]
Alright @earchibald-lv; I have a fix queued up for release on `main`. As profiles and chains can get very complicated, I'd like to ask for your help in testing it, if you're willing.

To perform a test against your own configuration, you can clone this repository and follow the setup instructions in [CONTRIBUTING.md](https://github.com/xeger/awsume-1password-plugin/blob/main/CONTRIBUTING.md) to create an isolated testing environment (it won't harm your real `awsume` or your Python installation). The test plan to execute is:

1. enter virtual-env shell
2. assume desired target profile; 1Password should provide an MFA token
3. exit and re-enter virtual-env shell (to remove AWS creds from the environment)
4. assume desired profile again; 1Password should _not_ be invoked this time

The assumption I'm making is that when an MFA token is required, it's always required for the first account in the role-assumption chain, never for accounts that come later in the chain. Thus, I lazily only check for a cached session on the _first_ profile, and avoid calling 1Password if it seems that no token will be required.

This simplification/assumption won't work for everyone, but AWSume itself has a fundamental limitation, as far as I can tell: only one MFA token can be passed on the CLI. So, there is a fundamental assumption that only one account in the chain will ever _need_ MFA. I'm hoping that for most people, it's the first.

If your testing indicates that this change is solid for you, I'll release a new version of the package. If your config is more complicated than mine, then I'll need to query you and find another solution.

Let me know how it goes. Thanks!

**Comment by xeger** [comment]
Hello @earchibald-lv -- I've published [v1.2.0](https://github.com/xeger/awsume-1password-plugin/releases/tag/v1.2.0) with a fix for your issue. Please let me know if it worked.
